source/extensions/transport_sockets/s2a/config.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "envoy/server/transport_socket_config.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace S2A {
+
+// S2A config registry
```

Comment:
Could you add some more comments about the role of the config registry?

Similarly for `UpstreamS2ATransportSocketConfigFactory` and `DownstreamS2ATransportSocketConfigFactory`?